docs/source/user/5minguide.rst

```diff
@@ -0,0 +1,247 @@
+.. _numba-5_mins:
+
+A ~5 minute guide to Numba
+==========================
+
+Numba is a just-in-time compiler for Python that works best on code that uses
+NumPy arrays and functions, and loops. The most common way to use Numba is
+through its collection of decorators that can be applied to your functions to
+instruct Numba to compile them. When a call is made to a Numba decorated
+function it is compiled to machine code "just-in-time" for execution and all or
+part of your code can subsequently run at native machine code speed! 
+
+Out of the box Numba works with the following:
+
+* OS: Windows (32 and 64 bit), OSX and Linux (32 and 64 bit)
+* Architecture: x86, x86_64, ppc64le. Experimental on armv7l, armv8l (aarch64).
+* GPUs: Nvidia CUDA. Experimental on AMD ROC.
+* CPython
+* NumPy 1.10 - latest
+
+How do I get it?
+----------------
+Numba is available as a `conda <https://conda.io/docs/>`_ package for the 
+`Anaconda Python distribution <https://www.anaconda.com/>`_::
+
+  $ conda install numba
+
+Numba also has wheels available::
+
+  $ pip install numba
+
+Numba can also be compiled from source (`instructions <LINK>`_)
+
+Numba is often used as a core package so its dependencies are kept to an
+absolute minimum, however, extra packages can be installed as follows to provide
+additional functionality:
+
+* ``scipy`` - enables support for compiling ``numpy.linalg`` functions
+* ``colorama`` - enables support for color highlighting in backtraces/error
+  messages
+* ``pyyaml`` - enables configuration of Numba via a YAML config file.
+* ``icc_rt`` - allows the use of the Intel SVML (high performance short vector
+  math library, x86_64 only).
+
+Will Numba work for my code?
+----------------------------
+This depends on what your code "looks like", if your code is numerically
+orientated (does a lot of math), uses NumPy a lot and/or has a lot of loops,
+then Numba is often a good choice. In these examples we'll apply the most
+fundamental of Numba's JIT decorators, ``@jit``, to try and speed up some
+functions to demonstrate what works well and what does not.
+
+Numba works well on code that "looks like" this::
+
+    from numba import jit
+    import numpy as np
+
+    x = np.arange(100).reshape(10, 10)
+
+    @jit
```

Comment:
Although you discuss nopython mode below, I would *start* the example with nopython mode and then rephrase the section below to explain why you do this.  I'd rather show best general practices, and then explain when to break them, than vice versa.